tests/acceptance/features/apiSharingNg/sharedWithMe.feature

```diff
@@ -1617,3 +1617,285 @@ Feature: an user gets the resources shared to them
+                },
+                "@client.synchronize": {
+                  "type": "boolean",
+                  "enum": [false]
+                },
+                "remoteItem": {
+                  "type": "object",
+                  "required": [
+                    "createdBy",
+                    "eTag",
+                    "file",
+                    "id",
+                    "lastModifiedDateTime",
+                    "name",
+                    "parentReference",
+                    "permissions",
+                    "size"
+                  ],
+                  "properties": {
+                    "permissions": {
+                      "type": "array",
+                      "items": {
+                        "oneOf": [
+                          {
+                            "type": "object",
+                            "required": [
+                            "grantedToV2",
+                            "id",
+                            "invitation",
+                            "roles"
+                            ],
+                            "properties": {
+                              "grantedToV2": {
+                                "type": "object",
+                                "required": ["user"],
+                                "properties": {
+                                  "group": {
+                                    "type": "object",
+                                    "required": ["displayName", "id"],
+                                    "properties": {
+                                      "displayName": {
+                                        "type": "string",
+                                        "enum": ["Brian Murphy"]
+                                      }
+                                    }
+                                  }
+                                }
+                              }
+                            }
+                          },
+                          {
+                            "type": "object",
+                            "required": [
+                            "grantedToV2",
+                            "id",
+                            "invitation",
+                            "roles"
+                            ],
+                            "properties": {
+                              "grantedToV2": {
+                                "type": "object",
+                                "required": ["group"],
+                                "properties": {
+                                  "group": {
+                                    "type": "object",
+                                    "required": ["displayName", "id"],
+                                    "properties": {
+                                      "displayName": {
+                                        "type": "string",
+                                        "enum": ["grp1"]
+                                      }
```

Comment:
why oneOf don't we have fix behavior?

Reply:
The order of the array items is random. I think adding that won't affect anything.

Reply:
Better if you avoid using oneOf if possible if not then we have to use it by checking false positive cases. 
did you try allOf ?

Reply:
Please add the min and max count:
```diff
  "type": "array",
+ "minItems": 2,
+ "maxItems": 2,
```
this should make the assertion little bit more strict

Reply:
we must use `minItems` adn `maxItems` for `oneOf` as well

Reply:
@saw-jan yes agreed but @nirajacharya2 have a thorough check all of it.